### PR TITLE
update to go.mod to k8s 1.22

### LINKED ISSUE #234: upgrade controller-runtime library

### PR DESCRIPTION
**Pull request by AdheipSingh**
Fixes #234.

### Description

<hr>

This PR has:
- [x] been tested on a real K8S cluster to ensure creation of a brand new Druid cluster works.
- [x] been tested for backward compatibility on a real K*S cluster by applying the changes introduced here on an existing Druid cluster. If there are any backward incompatible changes then they have been noted in the PR description.
- [ ] added comments explaining the "why" and the intent of the code wherever would not be obvious for an unfamiliar reader.
- [ ] added documentation for new or modified features or behaviors.

<hr>

### CODE REVIEW

File: examples/tiny-cluster.yaml [hunk 1]
```diff
@@ -167,8 +163,8 @@ spec:
         druid.processing.numMergeBuffers=1
         druid.processing.numThreads=1
         # Segment storage
-        druid.segmentCache.locations=[{\"path\":\"/druid/data/segments\",\"maxSize\":10737418240}]
-        druid.server.maxSize=10737418240
+        druid.segmentCache.locations=[{\"path\":\"/druid/data/segments\",\"maxSize\":10000000000}]
+        druid.server.maxSize=10000000000
```
Comment (nishantmonu51) [inline examples/tiny-cluster.yaml:167]: unrelated change?

Comment (AdheipSingh) [reply examples/tiny-cluster.yaml:167]: will revert this

**Comment by AdheipSingh** [comment]
cc @himanshug 
cut a release once this is merged

**Review by nishantmonu51 — APPROVED** [review]
👍
